presto-spi/src/main/java/com/facebook/presto/spi/relation/ExpressionOptimizer.java

```diff
@@ -25,12 +25,16 @@
     enum Level
     {
         /**
-         * SERIALIZABLE guarantees the optimized RowExpression can be serialized and deserialized
+         * SERIALIZABLE guarantees the optimized RowExpression can be serialized and deserialized and no type change in referenced input.
          */
         SERIALIZABLE,
         /**
          * MOST_OPTIMIZED removes all redundancy in a RowExpression but can end up with non-serializable objects (e.g., Regex).
          */
-        MOST_OPTIMIZED
+        MOST_OPTIMIZED,
```

Comment:
I found myself not naming this very well. Maybe we should call them `SERIALIZABLE`, `OPTIMAL`, and `EVALUATED`